front/src/app/account/new-order/new-order.component.ts

```diff
@@ -251,24 +256,33 @@ export class NewOrderComponent implements OnInit, OnDestroy {
 
     }
 
-    this.dataSource = new MatTableDataSource(order.items);
-
     for (const attr in this.orderItemFormGroup.controls) {
-      if (this.orderItemFormGroup.controls[attr]) {
-        this.orderItemFormGroup.removeControl(attr);
+      if (attr !== 'formatsForAll') {
+        if (this.orderItemFormGroup.controls[attr]) {
+          this.orderItemFormGroup.removeControl(attr);
+        }
       }
     }
 
-    order.items.forEach((item) => {
+    this.allAvailableFormats = new Set();
+    for (const item of order.items) {
+      item.available_formats?.forEach(format => this.allAvailableFormats.add(format));
```

Comment:
@kalbermattenm , here, for each order_item, add all available_formats to the allAvailableFormats Set